AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1901,9 +1912,13 @@ - (ASLayout *)calculateLayoutThatFits:(ASSizeRange)constrainedSize
     layoutSpec.isMutable = NO;
     ASLayout *layout = [layoutSpec measureWithSizeRange:constrainedSize];
     // Make sure layoutableObject of the root layout is `self`, so that the flattened layout will be structurally correct.
-    if (layout.layoutableObject != self) {
+    BOOL isRootLayout = (layout.layoutableObject != self);
+    if (isRootLayout) {
```

Comment:
Yeah I think the original code here was meant to make it the case that the layoutableObject is self, even though it isn't when returned by the layout spec.  The other case might occur in cases like returning a new node in finalLayoutable like is done with the layout visualizer or certain BRIO elements?